exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java

```diff
@@ -0,0 +1,532 @@
+      container.add(vector);
+      MapVector outputMapVector = (MapVector) vector;
+
+      for (ValueVector vv : mapVector) {
+        String fieldName = vv.getField().getLastName();
+        if (!keyList.contains(fieldName)) {
+          throw new UnsupportedOperationException("Unpivot data vector " +
+              ds + " contains key " + fieldName + " not contained in key source!");
+        }
+        if (vv.getField().getType().getMinorType() == TypeProtos.MinorType.MAP) {
+          throw new UnsupportedOperationException("Unpivot of nested map is not supported!");
+        }
+        if (field.equals("column")) {
+          outputMapVector.addOrGet(fieldName, vv.getField().getType(), vv.getClass());
+        } else {
+          List<LogicalExpression> args = Lists.newArrayList();
+          //TODO: Something else to access value of col such as emp_id?
+          args.add(SchemaPath.getSimplePath(vv.getField().getPath()));
+          //TODO: Put in the mapVector
+          LogicalExpression call = FunctionCallFactory.createExpression(functions.get(field), args);
+          //TODO: Is this sufficient to add to new Map?
+          ValueVector vector1 = addMapVector(fieldName, outputMapVector, call);
+          if (collector.hasErrors()) {
+            throw new SchemaChangeException("Failure while materializing expression. "
+                + collector.toErrorString());
+          }
+        }
+      }
+      dataSrcVecMap.put(ds, outputMapVector);*/
+    }
+    //Now create NDV in the outgoing container which was not avaliable in the incoming
+    for (VectorWrapper<?> vw : incoming) {
+      if (vw.getField().getLastName().equalsIgnoreCase("sum_width")) {//NullableFloat8 type
+        addVectorToOutgoingContainer("ndv", vw, collector);
+        break;
+      }
+    }
+    container.setRecordCount(0);
+    recordCount = 0;
+    container.buildSchema(incoming.getSchema().getSelectionVectorMode());
+  }
+
+  private void addVectorToOutgoingContainer(String field, VectorWrapper vw, ErrorCollector collector)
+     throws SchemaChangeException {
+    // Input map vector
+    MapVector mapVector = (MapVector) vw.getValueVector();
+    MaterializedField mf;
+    assert mapVector.getPrimitiveVectors().size() > 0;
+    // Proceed to create output map vector with same name e.g. statcount etc.
+    MajorType mt = mapVector.getField().getType();
+    if (functions.get(field) != null) {
+      mf = MaterializedField.create(functions.get(field), mt);
+    } else {
+      mf = MaterializedField.create(field, mt);
+    }
+    assert !dataSrcVecMap.containsKey(mf);
+    ValueVector vector = TypeHelper.getNewVector(mf, oContext.getAllocator());
+    container.add(vector);
+    MapVector outputMapVector = (MapVector) vector;
+
+    for (ValueVector vv : mapVector) {
+      String fieldName = vv.getField().getLastName();
+      if (!keyList.contains(fieldName)) {
+        throw new UnsupportedOperationException("Unpivot data vector " +
```

Comment:
`IllegalStateException` Actually, very confused about why the code even gets into this state: didn't we already check the names earlier? Isn't the code itself deciding to add the name? So, one part of the code is catching bugs in another part?

Maybe use `assert` or `Preconditions` if we are just checking for bugs?